src/textEditor.ts

```diff
@@ -107,5 +107,54 @@ export class TextEditor {
     static isLastLine(position : vscode.Position): boolean {
         return position.line === (vscode.window.activeTextEditor.document.lineCount - 1);
     }
+
+    static getLeftIndentation(line: string): number {
+        let tabSize = vscode.workspace.getConfiguration("editor").get<number>("tabSize");
+        let firstNonWhiteSpace = line.match(/^\s*/)[0].length;
```

Comment:
I see that when I operate on Position but here we are operating on string most of the time and this functionality is real simple, so I just moved it here. Maybe we can write some utility functions to handle all these kind of string matching?